velox/type/tests/TimestampConversionTest.cpp

```diff
@@ -104,6 +104,39 @@ TEST(DateTimeUtilTest, fromDateInvalid) {
       1970, 6, 31, "Date out of range: 1970-6-31"));
 }
 
+TEST(DateTimeUtilTest, fromWeekOfMonthDate) {
+  auto daysSinceEpochFromWeekOfMonthDate =
+      [](int32_t year, int32_t month, int32_t weekOfMonth, int32_t dayOfWeek) {
+        auto result = util::daysSinceEpochFromWeekOfMonthDate(
+            year, month, weekOfMonth, dayOfWeek);
+        EXPECT_TRUE(!result.hasError());
+        return result.value();
+      };
+  EXPECT_EQ(4, daysSinceEpochFromWeekOfMonthDate(1970, 1, 2, 1));
+  EXPECT_EQ(361, daysSinceEpochFromWeekOfMonthDate(1971, 1, 1, 1));
+  EXPECT_EQ(396, daysSinceEpochFromWeekOfMonthDate(1971, 2, 1, 1));
+
+  EXPECT_EQ(10952, daysSinceEpochFromWeekOfMonthDate(2000, 1, 1, 1));
+  EXPECT_EQ(19905, daysSinceEpochFromWeekOfMonthDate(2024, 7, 1, 1));
+
+  // Before unix epoch.
+  EXPECT_EQ(-3, daysSinceEpochFromWeekOfMonthDate(1970, 1, 1, 1));
+  EXPECT_EQ(-2, daysSinceEpochFromWeekOfMonthDate(1970, 1, 1, 2));
+  EXPECT_EQ(-31, daysSinceEpochFromWeekOfMonthDate(1969, 12, 1, 1));
+  EXPECT_EQ(-367, daysSinceEpochFromWeekOfMonthDate(1969, 1, 1, 1));
+  EXPECT_EQ(-724, daysSinceEpochFromWeekOfMonthDate(1968, 1, 2, 1));
+  EXPECT_EQ(-719533, daysSinceEpochFromWeekOfMonthDate(0, 1, 1, 1));
+
+  // Negative year - BC.
+  EXPECT_EQ(-719561, daysSinceEpochFromWeekOfMonthDate(-1, 12, 1, 1));
```

Comment:
I wonder if it would be easier to maintain these tests if we use date strings for 'expected' value.

```
EXPECT_EQ(parseDate("-1-12-01"), daysSinceEpoch(-1, 12, 1, 1)
EXPECT_EQ(parseDate("2024-03-05"), daysSinceEpoch(2024, 2, 5, 5));
```

Reply:
Thanks for your suggestion. In this way, if the test fails, it could be an issue with `parseDate`. And I'm not sure the behavior of `parseDate` is same as daysSinceEpochFromWeekOfMonthDate.